### PR TITLE
[WIP] Code refactoring using StyleCop.

### PR DESCRIPTION
**Pull request by kegor**

### CODE REVIEW

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423&bid=19152119)

File: src/Products/Common/Resources/Resources.cs [hunk 1]
```diff
@@ -43,41 +43,45 @@ public static string GetFreeFileName(string directory, string fileName)
...
+        /// Generate exception for password error.
         /// </summary>
-        /// <param name="ex">Exception</param>
-        /// <param name="password">string</param>
-        /// <returns>ExceptionEntity</returns>
-        public ExceptionEntity GenerateException(System.Exception ex, String password)
+        /// <param name="ex">Exception.</param>
+        /// <param name="password">string.</param>
+        /// <returns>ExceptionEntity.</returns>
+        public ExceptionEntity GenerateException(System.Exception ex, string password)
```
Comment (bobkovalex) [inline src/Products/Common/Resources/Resources.cs:72]: Codacy found an issue: [Make 'GenerateException' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850424&bid=19152120)

File: src/Products/Viewer/Controllers/ViewerApiController.cs [hunk 2]
```diff
@@ -1,47 +1,53 @@
...
+    /// ViewerApiController.
     /// </summary>
     [EnableCors(origins: "*", headers: "*", methods: "*")]
     public class ViewerApiController : ApiController
     {
-        private static Common.Config.GlobalConfiguration globalConfiguration;
+        /// <summary>
+        /// Map for locking keys in viewer cache.
+        /// </summary>
+        public static readonly ConcurrentDictionary<string, object> KeyLockerMap = new ConcurrentDictionary<string, object>();
```
Comment (bobkovalex) [inline src/Products/Viewer/Controllers/ViewerApiController.cs:41]: Codacy found an issue: [Use an immutable collection or reduce the accessibility of the field(s) 'KeyLockerMap'.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

File: src/Products/Viewer/Controllers/ViewerApiController.cs [hunk 3]
```diff
@@ -590,128 +693,139 @@ private static LoadDocumentEntity GetLoadDocumentEntity(GroupDocs.Viewer.Viewer
...
         }
 
-        private static string GetPageContent(Page page, string password, string documentGuid, ViewerSettings settings)
+        /// <summary>
+        /// Gets document pages data, dimensions and rotation angles.
+        /// </summary>
+        /// <param name="postedData">Posted data with document guid.</param>
+        /// <param name="loadAllPages">Flag to load all pages.</param>
+        /// <returns>Document pages data, dimensions and rotation angles.</returns>
+        private LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData, bool loadAllPages)
```
Comment (bobkovalex) [inline src/Products/Viewer/Controllers/ViewerApiController.cs:770]: Codacy found an issue: [Make 'GetDocumentPages' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423&bid=19152119)

File: src/Products/Common/Resources/Resources.cs (hunk 1 above)
Comment (bobkovalex) [inline src/Products/Common/Resources/Resources.cs:72]: Codacy found an issue: [Make 'GenerateException' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

File: src/Products/Viewer/Controllers/ViewerApiController.cs [hunk 4]
```diff
@@ -590,128 +689,139 @@ private static LoadDocumentEntity GetLoadDocumentEntity(GroupDocs.Viewer.Viewer
...
```
Comment (bobkovalex) [inline src/Products/Viewer/Controllers/ViewerApiController.cs:766]: Codacy found an issue: [Make 'GetDocumentPages' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Viewer/Controllers/ViewerApiController.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850424&bid=19152120)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- src/Products/Viewer/Controllers/ViewerApiController.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423&bid=19152119)

File: src/Products/Common/Resources/Resources.cs (hunk 1 above)
Comment (bobkovalex) [inline src/Products/Common/Resources/Resources.cs:72]: Codacy found an issue: [Make 'GenerateException' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

File: src/Products/Viewer/Controllers/ViewerApiController.cs [hunk 5]
```diff
@@ -521,197 +602,203 @@ private static ViewerSettings GetViewerSettings(string documentGuid, out string
...
         }
 
-        private static PageDescriptionEntity GetPageDescriptionEntities(Page page, string pagesInfoPath)
+        /// <summary>
+        /// Gets document pages data, dimensions and rotation angles.
+        /// </summary>
+        /// <param name="postedData">Posted data with document guid.</param>
+        /// <param name="loadAllPages">Flag to load all pages.</param>
+        /// <returns>Document pages data, dimensions and rotation angles.</returns>
+        private LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData, bool loadAllPages)
```
Comment (bobkovalex) [inline src/Products/Viewer/Controllers/ViewerApiController.cs:679]: Codacy found an issue: [Make 'GetDocumentPages' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Viewer/Controllers/ViewerApiController.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850424&bid=19152120)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Products/Viewer/Controllers/ViewerApiController.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423&bid=19152119)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Viewer/Controllers/ViewerApiController.cs  3
- src/Products/Viewer/Cache/MemoryPageStreamFactory.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850424&bid=19152120)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Products/Viewer/Cache/MemoryPageStreamFactory.cs  1
- src/Products/Viewer/Controllers/ViewerApiController.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423&bid=19152119)

**Pull request merged**